### PR TITLE
exporter/prometheus_remote_write: sort labels before exporting

### LINKED ISSUE #2920: Trouble writing metrics on thanos receiver
### Describe your environment

OS: Ubuntu: **22.04**
Python version: **Python 3.12.3**
Package version:
- opentelemetry-api==1.27.0
- opentelemetry-sdk==1.27.0
- opentelemetry-exporter-prometheus-remote-write==0.48b0

### What happened?

I write a simple code to create a gauge metric on thanos receiver, but the operation return the error:

_Export POST request failed with reason: 409 Client Error: Conflict for url: <thanos receiver andpoint>_

In the log of the receiver I observed thi error: 

_msg="failed to handle request" err="add 1 series: out of order labels"_

### Steps to Reproduce

Execute this code with a working thanos receiver https uri:

```
from opentelemetry.exporter.prometheus_remote_write import PrometheusRemoteWriteMetricsExporter
from opentelemetry.metrics import Observation, set_meter_provider, get_meter
from opentelemetry.sdk.metrics import MeterProvider
from opentelemetry.sdk.metrics.export import PeriodicExportingMetricReader

name = 'test_metric'
desc = 'test_metric'
unit = 'count'
value = 1000
attributes = {'job': 'testJob'}

exporter = PrometheusRemoteWriteMetricsExporter(
    endpoint=<thanos receiver https uri>,
    tls_config={'insecure_skip_verify': False},
    resources_as_labels=True
)
reader = PeriodicExportingMetricReader(exporter, 1000)
provider = MeterProvider(metric_readers=[reader])
set_meter_provider(provider)
meter = get_meter(__name__)
gauge = meter.create_gauge(name, description=desc, unit=unit)
gauge.set(value, attributes)
```

### Expected Result

Metric is created in thanos.

### Actual Result

Metric is rejected from thanos.

### Additional context

I open the code of the class **PrometheusRemoteWriteMetricsExporter** and make a little modification in method **_convert_to_timeseries** ordering the label before create the timeseries.

This is my working version of the method:

```
def _convert_to_timeseries(
    self, sample_sets: Sequence[tuple], resource_labels: Sequence
) -> Sequence[TimeSeries]:
    timeseries = []
    for labels, samples in sample_sets.items():
        ts = TimeSeries()
        labels = list(labels) + list(resource_labels)
        labels.sort(key=lambda x: x[0])
        for label_name, label_value in chain(labels):
            ts.labels.append(self._label(label_name, str(label_value)))
        for value, timestamp in samples:
            ts.samples.append(self._sample(value, timestamp))
        timeseries.append(ts)
    return timeseries
```

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by xrmx**
# Description

Sort labels lexicographically before exporting them.  

Co-authored-by: fstolba / fst0

Fixes #2920
Closes #2784

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] tox

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

File: exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py [hunk 1]
```diff
@@ -253,12 +253,14 @@ def _parse_metric(
         return self._convert_to_timeseries(sample_sets, resource_labels)
 
     def _convert_to_timeseries(
-        self, sample_sets: Sequence[tuple], resource_labels: Sequence
+        self, sample_sets: Mapping[tuple, Sequence], resource_labels: Sequence
```
Comment (tammy-baylis-swi) [inline exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py:256]: Just curious: how come this signature change is needed, compared to https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2784 ? Was `Sequence[tuple]` the wrong type before?

**Review by emdneto — APPROVED** [review]

**Review by tammy-baylis-swi — APPROVED** [review]
Lgtm, just a question.

**Review by lzchen — APPROVED** [review]

File: exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py (hunk 1 above)
Comment (xrmx) [reply exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py:256]: Yes, it's initialized as defaultdict(list) in the only caller:
https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2940/files/68062d8256a28909cfc458aaaa77748f7f19f525#diff-4be6a764cd546e2cea72c7e3f029a9170e541fd7480b5df96f111e6862576879R240

Comment (tammy-baylis-swi) [reply exporter/opentelemetry-exporter-prometheus-remote-write/src/opentelemetry/exporter/prometheus_remote_write/__init__.py:256]: Nice, thanks!

**Pull request merged**